SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -92,6 +104,87 @@ void USpatialWorkerConnection::Connect(bool bInitAsClient)
 	}
 }
 
+#pragma optimize("", off)
+void USpatialWorkerConnection::OnLoginTokens(void* UserData, const Worker_Alpha_LoginTokensResponse* LoginTokens)
+{
+	if (LoginTokens->status.code != WORKER_CONNECTION_STATUS_CODE_SUCCESS)
+	{
+		UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Failed to get login token, StatusCode: %d, Error: %s"), LoginTokens->status.code, UTF8_TO_TCHAR(LoginTokens->status.detail));
+		return;
+	}
+
+	if (LoginTokens->login_token_count == 0)
+	{
+		UE_LOG(LogSpatialWorkerConnection, Warning, TEXT("No deployment found to connect to. Did you add the 'dev_login' tag to the deployment you want to connect to?"));
+		return;
+	}
+
+	UE_LOG(LogSpatialWorkerConnection, Log, TEXT("Success: Login Token Count %d"), LoginTokens->login_token_count);
+	USpatialWorkerConnection* Connection = static_cast<USpatialWorkerConnection*>(UserData);
+	const FString& DeploymentToConnect = GetDefault<USpatialGDKSettings>()->DevelopmentDeploymentToConnect;
+	// If not set, use the first deployment.
+	if (DeploymentToConnect.IsEmpty())
+	{
+		Connection->LocatorConfig.LoginToken = FString(LoginTokens->login_tokens[0].login_token);
```

Comment:
would add a comment here as well saying that the first login token that you get can change with every query, because they are not ordered